ecs/src/nox/ecs/component/Transform.h

```diff
@@ -0,0 +1,84 @@
+#ifndef NOX_ECS_TRANSFORM_H_
+#define NOX_ECS_TRANSFORM_H_
+
+#include <string>
+
+#include <nox/ecs/Component.h>
+#include <nox/ecs/EntityManager.h>
+
+#include <glm/mat4x4.hpp>
+#include <glm/vec2.hpp>
+
+
+
+namespace nox
+{
+    namespace ecs
+    {
+        /**
+         * Place an Actor with a position, rotation, and scale.
+         * An important component used by many other components.
+         *
+         * # JSON Description
+         * ## Name
+         * %Transform
+         *
+         * ## Properties
+         * - __position__:vec2 - Position of the Actor. Default vec2(0, 0).
+         * - __rotation__:real - Rotation of the Actor. Default 0.
+         * - __scale__:vec2 - Scale of the actor. Default vec2(1, 1).
+         */
+        class Transform final
+            : public Component
+        {
+        public:
+            const static std::string NAME;
+        
+            virtual ~Transform() = default;
```

Comment:
Destructor should not be virtual, we are not doing virtual inheritance, and this class is final. This will just add the vptr table, which we are not interested in. Also the destructor is not needed.